sdk/storage/azblob/blockblob/models.go

```diff
@@ -70,6 +70,59 @@ func (o *UploadOptions) format() (*generated.BlockBlobClientUploadOptions, *gene
 
 // ---------------------------------------------------------------------------------------------------------------------
 
+// UploadBlobFromURLOptions contains the optional parameters for the Client.UploadBlobFromURL method.
+type UploadBlobFromURLOptions struct {
+	// Optional. Used to set blob tags in various blob operations.
+	Tags map[string]string
+
+	// Only Bearer type is supported. Credentials should be a valid OAuth access token to copy source.
+	CopySourceAuthorization *string
```

Comment:
How is this used?

Reply:
It's an optional parameter that uses bearer token for authorization.

Reply:
Do we expect customers to populate this with the actual bearer token string, or should this be an `azcore.TokenCredential`?  What do other languages do?

Reply:
@jhendrixMSFT I looked into the other languages, so for Java and .NET, they have a special HttpAuthorization type provided by Azure Core
https://learn.microsoft.com/en-us/dotnet/api/azure.httpauthorization.-ctor?view=azure-dotnet#azure-httpauthorization-ctor(system-string-system-string)
https://azuresdkartifacts.blob.core.windows.net/azure-sdk-for-java/staging/apidocs/com/azure/core/http/HttpAuthorization.html

For Python, looks like it's just a string https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py#L583

Reply:
I see.  Is it the case that we're authenticating with an endpoint outside of Azure?

Reply:
https://learn.microsoft.com/en-us/rest/api/storageservices/put-blob-from-url?tabs=azure-ad#request-headers

> Only scheme bearer is supported for Azure Active Directory.

Reply:
@jhendrixMSFT, `CopySourceAuthorization` is a `*string` everywhere else in AzBlob.  For example - https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/storage/azblob/blockblob/models.go#L100.  Put Block from URL also only supports authenticating with AAD - https://learn.microsoft.com/en-us/rest/api/storageservices/put-block-from-url?tabs=azure-ad#request-headers